### PR TITLE
++Add Payment Class Methods and Update Dependencies

### PR DESCRIPTION
**Pull request by tlacloc**
++

## Summary by Sourcery

Add new methods to the Stripe service for customer and tax management, refactor authentication and payment logic into controllers, and enhance the user model with additional fields. Update dependencies and expand test coverage for the new features.

New Features:
- Introduce new methods in the Stripe service for customer management, including creating, retrieving, updating, and deleting customers.
- Add functionality to calculate and retrieve tax settings for payment intents.
- Implement a new IP descriptor service to fetch IP details and validate IP addresses.
- Add new GraphQL mutations and queries for managing user and course data, including registering customer IDs and retrieving course details by ID.

Enhancements:
- Refactor the authentication and payment logic into separate controllers for better modularity and maintainability.
- Update the Stripe API version and include beta features for payment intents with tax calculations.
- Enhance the user model with additional fields for country and customer ID, and add triggers for automatic timestamp updates.
- Refactor the login and signup processes to include IP address handling for Stripe customer registration.

Build:
- Update dependencies to include new packages such as geoip-lite and axios for IP handling and HTTP requests.

Tests:
- Expand test coverage for the Stripe service to include customer operations and tax calculations.
- Refactor existing tests to use the new testing framework and assertion library.

Chores:
- Rename model classes to follow a consistent naming convention with 'Model' suffix.

### CODE REVIEW

**Comment by tlacloc** [comment]
@sourcery-ai review

File: api/model/Auth.js [hunk 1]
```diff
@@ -7,15 +7,29 @@ const { verifySignature } = require('@taquito/utils')
...
-  constructor ({ gql, jwt, email, opts }) {
+  constructor ({ gql, jwt, email, opts, stripe }) {
     this.gql = gql
     this.opts = opts
     this.email = email
     this.jwt = jwt
+    this.stripe = stripe
+  }
+
+  async stripeRegister (user, ip) {
```
[the remark this answers is not in the document]
Comment (tlacloc) [reply api/model/Auth.js:23]: It just loggs the error to not stop the user login

**Comment by tlacloc** [comment]
@sourcery-ai review

File: api/controller/payments.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    this.userModel = userModel
+    this.courseModel = courseModel
+  }
+
+  async stripeWebhook (req, res) {
+    const sig = req.headers['stripe-signature']
+    const response = await this.payments.verifyStripeWebhook(sig, req.rawBody)
+    req.log.info(`stripe verify response:  ${JSON.stringify(response)}`)
+
+    return response
```
Comment (crissthiandi) [inline api/controller/payments.js:13]: this.payments es paymentsModel o reconoce la clase de payments

Comment (tlacloc) [reply api/controller/payments.js:13]: Es el model, gracias!

File: api/graphQL/users.js [hunk 3]
```diff
@@ -117,6 +123,27 @@ mutation (
...
+    country: $country
+  }) {
+    id
+    customer_id
+    country
+  }
+}
+`
+
 const UPDATE_USER_PASSWORD = gql`
```
Comment (crissthiandi) [inline api/graphQL/users.js:147]: identación?

**Review by crissthiandi — APPROVED** [review]
LGTM

**Pull request merged**